content/general/concepts/what-is-graphql/what-is-graphql.md

```diff
@@ -0,0 +1,38 @@
+---
+Title: 'What is GraphQL?' # This is the name that appears on the hub page for this language. Pay attention to capitalization and punctuation!
+Description: 'GraphQL is an open-source query language.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'APIs'
+  - 'Servers'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/back-end-engineer-career-path'
+  - 'paths/front-end-engineer-career-path'
+---
+
+## History of GraphQL
+
+GraphQL was developed by Facebook in 2012 to address internal issues with their mobile applications. As an alternative to typical REST architecture, GraphQL became open-source in 2015. Consequently, GraphQL makes data transfers more efficient, network usage for Facebook's mobile applications was drastically reduced. As of now, GraphQL has attracted a large number of developers and is becoming increasingly popular for client-server communication.
+
+## What is GraphQL?
+
+GraphQL, an open-source query language as described in [GraphQL](graphql.org)'s official website. It allows developers to consume, construct, and create APIs in their own unique way.
+
+## Why adopt GraphQL?
+
+Before discussing GraphQL, developers should be familiar with REST. One of REST's disadvantages is that it can result in an excessive number of HTTP queries or too much data being fetched. GraphQL aims to improve upon REST philosophy by allowing developers to ask for a specific data instead of fetching a whole giant object. There is no room for errors like underfetching and overfetching with GraphQL. Finally, GraphQL allows developers to easily integrate with databases, JSON files, REST APIs and cloud services.
+
+## How GraphQL works under the hood?
+
+A typical yet simple GraphQL workflow can be seen below:
```

Comment:
> A typical yet simple GraphQL workflow can be seen below:

I advise against using words like "simple" and "basic". We could, instead, use "common" to retain the spirit of "typical" and showcase beginner-friendliness as opposed to "simple".